js/src/admin/components/MailPage.js

```diff
@@ -68,65 +69,69 @@ export default class MailPage extends Page {
             <h2>{app.translator.trans('core.admin.email.heading')}</h2>
             <div className="helpText">{app.translator.trans('core.admin.email.text')}</div>
 
-            {FieldSet.component({
-              label: app.translator.trans('core.admin.email.addresses_heading'),
-              className: 'MailPage-MailSettings',
-              children: [
+            <FieldSet label={app.translator.trans('core.admin.email.addresses_heading')} className="MailPage-MailSettings">
+              <div className="MailPage-MailSettings-input">
+                <label>
+                  {app.translator.trans('core.admin.email.from_label')}
+                  <input className="FormControl" value={this.values.mail_from() || ''} oninput={m.withAttr('value', this.values.mail_from)} />
+                </label>
+              </div>
+            </FieldSet>
+
+            <FieldSet label={app.translator.trans('core.admin.email.driver_heading')} className="MailPage-MailSettings">
+              <div className="MailPage-MailSettings-input">
+                <label>
+                  {app.translator.trans('core.admin.email.driver_label')}
+                  <Select
+                    value={this.values.mail_driver()}
+                    options={Object.keys(this.driverFields).reduce((memo, val) => ({ ...memo, [val]: val }), {})}
+                    onchange={this.values.mail_driver}
+                  />
+                </label>
+              </div>
+            </FieldSet>
+
+            {this.status.sending
+              ? ''
+              : Alert.component({
+                  children: app.translator.trans('core.admin.email.not_sending_message'),
+                  dismissible: false,
+                })}
+
+            {fieldKeys.length > 0 ? (
```

Comment:
This could be `{!!fieldKeys.length && (...)}` (not sure if the `!!` is needed to make sure the `0` doesn't render - probably not, feel free to test.